### ISSUE TITLE
register fails when egrep doesn't support -a

### ISSUE THREAD

**Issue by bahamat** [dehydrated-io/dehydrated#839]
The detection for which grep to use doesn't take into account that -a is not always supported, which leads to it not setting `GREP=awk_egrep`.

```
# ./dehydrated --register --accept-terms
# INFO: Using main config file /opt/dehydrated/config
egrep: illegal option -- a
usage: egrep [-bchHilLnoqrRsvx] [-A num] [-B num] [-C num|-num]
             [--label=name] [-e pattern_list]... [-f pattern_file]...
             [pattern_list] [file]...
EXPECTED value GOT EOF
```

Additionally, systems where egrep doesn't support `-a`, probably won't have `gawk`. But even if they do, gawk produces this error:

```
gawk: cmd. line:3: (FILENAME=- FNR=1) warning: regexp escape sequence `\u' is not a known regexp operator
```

or

```
gawk: cmd. line:3: (FILENAME=- FNR=1) fatal: Invalid range end: /"[^[:cntrl:]"\]*((\[^u[:cntrl:]]|\u[0-9a-fA-F]{4})[^[:cntrl:]"\]*)*"|-?(0|[1-9][0-9]*)([.][0-9]*)?([eE][+-]?[0-9]*)?|null|false|true|[[:space:]]+|./
```

(depending on version).

**Comment by lukas2511** [comment]
Hey, thanks for the input. I actually have a version of dehydrated lurking on my system (for waaay to long..) that got rid of the egrep and gawk usage completely, but I didn't push it yet because I was kinda stuck on setting up a testing system, which I wanted to do before pushing code that could have broken dehydrated on various systems...

I've uploaded that version here (for now): https://gist.github.com/lukas2511/5baa4fd9358f5fa622b90f3cfb2c0a19

It would be great if you could check if it works for you.

**Comment by bahamat** [comment]
I'll try that one and let you know.

**Comment by bahamat** [comment]
Nope, that one doesn't. I got here originally because on illumos, our grep doesn't support `-a`, and that's the error I still get. here.

```
# ./dehydrated --register --accept-terms
# INFO: Using main config file /opt/dehydrated/config
grep: illegal option -- a
usage:  grep [-E|-F] [-bchHilLnoqrRsvx] [-A num] [-B num] [-C num|-num]
             [--label=name] [-e pattern_list]... [-f pattern_file]...
             [pattern_list] [file]...
EXPECTED value GOT EOF
```

**Comment by lukas2511** [comment]
Should hopefully be fixed by the merged pull-request (#840).
